crypto/crypto.go

```diff
@@ -16,23 +23,26 @@ type Encrypted struct {
 
 type Signed struct {
 	Message string
-	Mode    string
+	Mode    Mode
 	//Inputs map[string]string
 	Signature string
 }
 
-func NewSignature() *Signed {
-	return &Signed{Mode: SignatureMode}
-}
-
 func NewHMAC() *Signed {
 	return &Signed{Mode: HMACMode}
 }
 
+func NewSignature(mode Mode) *Signed {
+	return &Signed{Mode: mode}
+}
+
 func GroupEncrypt(plaintext string, publicKeys map[string]string) (*Encrypted, error) {
 
 	keySize := 32
-	key := RandomBytes(keySize)
+	key, err := RandomBytes(keySize)
+	if err != nil {
+		return nil, err
```

Comment:
Do we not need to add some context to the error? If not then we can probably remove a ton of fmt.Errorf()s throughout the whole project.